src/tests/ftest/telemetry/engine_events.py

```diff
@@ -0,0 +1,270 @@
+'''
+  (C) Copyright 2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+import time
+
+from dmg_utils import check_system_query_status
+from general_utils import report_errors
+from telemetry_test_base import TestWithTelemetry
+
+
+class EngineEvents(TestWithTelemetry):
+    """
+    Verify engine-related event values in telemetry.
+
+    :avocado: recursive
+    """
+
+    def collect_telemetry(self, rank_count):
+        """Collect the following engine event values.
+
+        engine_events_dead_ranks, engine_events_last_event_ts, engine_servicing_at, and
+        engine_servicing_at.
+
+        Args:
+            rank_count (int): Number of ranks in the system.
+
+        Returns:
+            list: Four lists. Each list contains the above telemetry value for each rank.
+
+        """
+        metric_to_data = {}
+        for metric in self.telemetry.ENGINE_EVENT_METRICS:
+            metric_to_data[metric] = self.telemetry.get_metrics(name=metric)
+        self.log.info("metric_to_data = %s", metric_to_data)
+
+        # Omit "engine" from the variable name for brevity. The indices correspond to ranks.
+        events_dead_ranks = [None for _ in range(rank_count)]
+        hosts = list(self.hostlist_servers)
+        for host in hosts:
+            engine_events_dead_ranks = metric_to_data["engine_events_dead_ranks"]
+            metrics = engine_events_dead_ranks[host]["engine_events_dead_ranks"]["metrics"]
```

Comment:
Is this correct? Combining these two lines, the data structure would be
```
metrics = metric_to_data["engine_events_dead_ranks"][host]["engine_events_dead_ranks"]
```
It seems `engine_events_dead_ranks` is nested under itself?

Reply:
Yes, `engine_events_dead_ranks` is nested under the key of the same name.

Reply:
Ah, I see now. It's because above does this
```
        for metric in self.telemetry.ENGINE_EVENT_METRICS:
            metric_to_data[metric] = self.telemetry.get_metrics(name=metric)
```

But it seems this should work too
```
        metric_to_data = self.telemetry.get_metrics(
            name=','.join(self.telemetry.ENGINE_EVENT_METRICS))
```

And then the structure would look something like
```
metric_to_data = {
    "host1": {
        "engine_events_dead_ranks": ...,
        "engine_events_last_event_ts": ...
    }
```


And this
```
            engine_events_dead_ranks = metric_to_data["engine_events_dead_ranks"]
            metrics = engine_events_dead_ranks[host]["engine_events_dead_ranks"]["metrics"]
```
Could become
```
            metrics = metric_to_data[host]["engine_events_dead_ranks"]["metrics"]
```

Reply:
Thanks! I'll try it in the separate PR.

Reply:
Done